heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraHeronShellController.java

```diff
@@ -0,0 +1,119 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.scheduler.aurora;
+
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import com.twitter.heron.proto.system.PhysicalPlans.StMgr;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.ConfigLoader;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.packing.PackingPlan;
+import com.twitter.heron.spi.statemgr.IStateManager;
+import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+import com.twitter.heron.spi.utils.NetworkUtils;
+import com.twitter.heron.spi.utils.ReflectionUtils;
+
+/**
+ * Implementation of AuroraController that is a wrapper of AuroraCLIController
+ * The difference is: the `restart` method implementation is changed to heron-shell
+ */
+class AuroraHeronShellController implements AuroraController {
+  private static final Logger LOG = Logger.getLogger(AuroraHeronShellController.class.getName());
+
+  private final String topologyName;
+  private final AuroraCLIController cliController;
+  private SchedulerStateManagerAdaptor stateMgrAdaptor;
+
+  AuroraHeronShellController(
+      String jobName,
+      String cluster,
+      String role,
+      String env,
+      String auroraFilename,
+      boolean isVerbose) {
+    this.topologyName = jobName;
+    this.cliController = new AuroraCLIController(
+        jobName, cluster, role, env, auroraFilename, isVerbose);
+
+    stateMgrAdaptor = null;
+    Config config =
+        Config.toClusterMode(Config.newBuilder().putAll(ConfigLoader.loadClusterConfig()).build());
+    String stateMgrClass = Context.stateManagerClass(config);
+    IStateManager stateMgr = null;
+    try {
+      stateMgr = ReflectionUtils.newInstance(stateMgrClass);
+
+      stateMgr.initialize(config);
+      stateMgrAdaptor = new SchedulerStateManagerAdaptor(stateMgr, 5000);
+    } catch (ClassNotFoundException e) {
+      LOG.severe(e.getMessage());
+    } catch (InstantiationException e) {
+      LOG.severe(e.getMessage());
+    } catch (IllegalAccessException e) {
+      LOG.severe(e.getMessage());
+    }
+  }
+
+  @Override
+  public boolean createJob(Map<AuroraField, String> bindings) {
+    return cliController.createJob(bindings);
+  }
+
+  @Override
+  public boolean killJob() {
+    return cliController.killJob();
+  }
+
+  // Restart an aurora container
+  @Override
+  public boolean restart(Integer containerId) {
+    if (containerId == null) {
+      throw new UnsupportedOperationException("Not implemented");
+    }
+
+    if (stateMgrAdaptor == null) {
+      LOG.warning("SchedulerStateManagerAdaptor not initialized");
+      return false;
+    }
+
+    StMgr contaienrInfo = stateMgrAdaptor.getPhysicalPlan(topologyName).getStmgrs(containerId);
+    String ip = contaienrInfo.getHostName();
```

Comment:
call the variable `host` or `hostname`. `ip` is confusing here.

Reply:
done